ext/stackprof/stackprof.c

```diff
@@ -224,6 +225,17 @@ frame_lines_i(st_data_t key, st_data_t val, st_data_t arg)
     return ST_CONTINUE;
 }
 
+static VALUE
+coerce_frame_name(VALUE name, VALUE line) {
+    char *start_pointer = strstr(RSTRING_PTR(name), "<top (required)>\0");
```

Comment:
Wouldn't some sort of `strncmp` be better here? Does the string match exactly, or is it sometimes contained within?

Reply:
It either match exactly or is trailing:

  - `<top (required)>`
  - `block in <top (required)>`